lib/Migration/LabelMismatchCleanup.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Deck\Migration;
+
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+
+class LabelMismatchCleanup implements IRepairStep {
+
+	public function __construct(
+		private IDBConnection $db,
+	) {
+	}
+
+	public function getName() {
+		return 'Migrate labels with wrong board mapping';
+	}
+
+	public function run(IOutput $output) {
+		// Find assingments where a label of another (wrong) board is used
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('al.id', 'al.label_id', 'al.card_id', 's.board_id as actual_board_id', 'l.board_id as wrong_id', 'l.color', 'l.title')
+			->from('deck_assigned_labels', 'al')
+			->innerJoin('al', 'deck_cards', 'c', 'c.id = al.card_id')
+			->innerJoin('c', 'deck_stacks', 's', 'c.stack_id = s.id')
+			->innerJoin('al', 'deck_labels', 'l', 'l.id = al.label_id')
+			->where($qb->expr()->neq('l.board_id', 's.board_id'));
+		
+		$labels = $qb->execute()->fetchAll();
```

Comment:
I think the `execute()` function is deprecated. We should use `executeQuery()` instead